src/meta/server_state.cpp

```diff
@@ -56,6 +56,24 @@ using namespace dsn;
 namespace dsn {
 namespace replication {
 
+DSN_DEFINE_int32("meta_server",
```

Comment:
```
DSN_DEFINE_uint32("meta_server",
                 max_allowed_replica_count,
                 3,
                 "max allowed replica count for arbitrary number of nodes in a cluster");

DSN_DEFINE_validator(max_allowed_replica_count, [](int32_t allowed_replica_count) -> bool {
    return allowed_replica_count <= 3;
});
```

Reply:
Could we keep the type of `FLAGS_ max_allowed_replica_count` consistent with `max_replica_count` whose type is `int32` ? So we can compare them without explicitly/implicitly conversion between `int32` and `uint32`.

On the other hand, could we reserve `allowed_replica_count  > 0` ? Since zero replica count is invalid.